testing/sdk_networks.py

```diff
@@ -44,16 +48,28 @@ def check_task_network(task_name, expected_network_name="dcos"):
                     )
 
 
+@retrying.retry(
+    wait_fixed=1000, stop_max_delay=5 * 1000, retry_on_result=lambda res: not res
+)
+def wait_for_endpoint_info(
+    package_name: str, service_name: str, endpoint_name
+) -> typing.Dict:
+    ret = sdk_cmd.svc_cli(
+        package_name, service_name, "endpoints {}".format(endpoint_name), json=True
+    )
+    return ret
+
+
 def get_and_test_endpoints(package_name, service_name, endpoint_to_get, correct_count):
     """Gets the endpoints for a service or the specified 'endpoint_to_get' similar to running
     $ docs <service> endpoints
     or
     $ dcos <service> endpoints <endpoint_to_get>
     Checks that there is the correct number of endpoints"""
-    endpoints = sdk_cmd.svc_cli(
-        package_name, service_name, "endpoints {}".format(endpoint_to_get), json=True
-    )
-    assert len(endpoints) == correct_count, "Wrong number of endpoints, got {} should be {}".format(
+    endpoints = wait_for_endpoint_info(package_name, service_name, endpoint_to_get)
+    assert (
+        len(endpoints) == correct_count
+    ), "Wrong number of endpoints, got {} should be {}".format(
         len(endpoints), correct_count
```

Comment:
While we're at it, we may want to print the list of endpoints in case of failure too.